pkg/block/node.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package block
+
+import (
+	"github.com/oklog/ulid"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+)
+
+// Node type implements a node of a Forest.
```

Comment:
forest still ):

Reply:
Suggestions? Node of a tree?

Reply:
Aren't both test cases relevant? The first one is for case where two blocks have same sources and second one is where the blocks have overlapping sources.